src/main/java/seedu/address/logic/commands/AddItemTagCommand.java

```diff
@@ -0,0 +1,111 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ITEM_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ITEM_TAG;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Logger;
+
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.logic.commands.EditItemCommand.EditItemDescriptor;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.item.Item;
+import seedu.address.model.tag.Tag;
+
+public class AddItemTagCommand extends Command {
+    public static final String COMMAND_WORD = "addt";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a set of tags to an item "
+            + "at the index number used in the displayed item list. "
+            + "Existing tags added will be ignored by the command.\n"
+            + "Parameters: "
+            + "[" + PREFIX_ITEM_NAME + "NAME] "
+            + "[" + PREFIX_ITEM_TAG + "TAG(s) TO ADD] "
+            + "Example: " + COMMAND_WORD + PREFIX_ITEM_NAME + "Iron "
+            + PREFIX_ITEM_TAG + "delicious, tuturu";
+
+    public static final String MESSAGE_ITEM_NOT_PROVIDED = "Item name was not provided.";
+    public static final String MESSAGE_ITEM_NOT_FOUND = "Item was not found in the item list.";
+    public static final String MESSAGE_TAG_NOT_ADDED = "No new tag is added.";
+    public static final String MESSAGE_TAG_NOT_PROVIDED = "Item Tag was not provided";
+
+    private static final Logger logger = LogsCenter.getLogger(AddItemTagCommand.class);
+
+    private final String itemName;
+    private final Set<Tag> tags;
+
+    /**
+     * Constructs AddItemTagCommand to execute command
+     * @param itemName item name to identify item in the list
+     * @param tags set of tags
+     */
+    public AddItemTagCommand(String itemName, Set<Tag> tags) {
+        requireNonNull(itemName);
+
+        this.itemName = itemName;
+        this.tags = new HashSet<>();
+        this.tags.addAll(tags);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        List<Item> itemList = new ArrayList<>(model.getFilteredItemList());
+
+        // filter to only get matching and not deleted items
+        itemList.removeIf(x -> !x.getName().equals(itemName));
+        if (itemList.isEmpty()) {
+            throw new CommandException(MESSAGE_ITEM_NOT_FOUND);
+        }
```

Comment:
Wouldn't it be better to set the filter to search for the item, then use the filteredList? Right now it seems like it uses the "currently filtered list" which might not be intended if you use this command after doing maybe a "find", as the item searched for might not exist in the subset filtered.

Reply:
this should work fine since the filters are reset at the start of every command